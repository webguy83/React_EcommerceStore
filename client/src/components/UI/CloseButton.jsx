import React from "react";
import styled from "styled-components/macro";

const CloseButtonContainer = styled.div`
  cursor: pointer;
  font-size: 2.9rem;

  &:hover,
  &:active {
    color: var(--sec);
    transition: var(--btnTrans);
  }
`;

const CloseButton = ({ closeElm }) => {
  return (
    <CloseButtonContainer data-testid="container" onClick={closeElm}>
      &times;
    </CloseButtonContainer>
  );
};

export default CloseButton;
