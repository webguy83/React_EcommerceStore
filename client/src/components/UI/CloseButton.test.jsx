import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CloseButton from "./CloseButton";
import "jest-styled-components";

afterEach(cleanup);

describe("Close Button default tests", () => {
  it("should render without crashing", () => {
    const { asFragment } = render(<CloseButton closeElm={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the X symbol for close button", () => {
    const { getByTestId } = render(<CloseButton closeElm={() => {}} />);
    expect(getByTestId("container")).toHaveTextContent(/^Ã—$/);
  });
});
