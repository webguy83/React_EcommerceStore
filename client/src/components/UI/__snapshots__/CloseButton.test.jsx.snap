// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Button default tests should render without crashing 1`] = `
<DocumentFragment>
  .c0 {
  cursor: pointer;
  font-size: 2.9rem;
}

.c0:hover,
.c0:active {
  color: var(--sec);
  -webkit-transition: var(--btnTrans);
  transition: var(--btnTrans);
}

<div
    class="c0"
    data-testid="container"
  >
    Ã—
  </div>
</DocumentFragment>
`;
